presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java

```diff
@@ -317,6 +347,33 @@ private synchronized QueryResults getNextResult(OptionalLong token, UriInfo uriI
             }
         }
 
+        // if query query submission has not finished, return simple empty result
+        if (!submissionFuture.isDone()) {
+            QueryResults queryResults = new QueryResults(
+                    queryId.toString(),
+                    uriInfo.getRequestUriBuilder().replaceQuery(queryId.toString()).replacePath("query.html").build(),
```

Comment:
This is incorrect after the `/ui` change. Move the `queryHtmlUri` from below so we only build the URL in one place.